### PR TITLE
fix: include user id to paswordReset token, modify endpoint to enhanc…

### PR DESCRIPTION
**Pull request by Jidekas**
…e security

**Description of PR**
fix: This PR includes userId passwordReset token and modifies the code to enhance security.

_A brief description of what changes your PR introduces. The better this description, the quicker we can review, exchange
feedback, and merge!_

---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [x] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ ] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [ ] Does the endpoint return the correct success/error format?

---

**Screenshots**  
_add relevant screenshots here_

---

**Test Coverage**  
_add relevant screenshots here_

### CODE REVIEW

**Comment by KristofaJosh** [comment]
Task linked: [CU-86bwj0akd BE - Auth - Forgot Password](https://app.clickup.com/t/86bwj0akd)

**Review by polymathmode — APPROVED** [review]
nice job,well done!

File: apps/server/src/controllers/userControllers/authUsers.ts [hunk 1]
```diff
@@ -247,10 +247,14 @@ export const forgotPassword = async (req: Request, res: Response) => {
...
-          otp: otp.toString(),
-          tokenExpiry: tokenExpiry.getTime().toString(),
-        })
+        const token = await Jwt.sign(
+          {
+            id: user.id,
+            otp: otp.toString(),
+            tokenExpiry: tokenExpiry.getTime().toString(),
+          },
+          { _secret: Env.JWT_SECRET + longString }
```
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/authUsers.ts:256]: if you do this, how do you intend verifying the token?

Comment (KristofaJosh) [reply apps/server/src/controllers/userControllers/authUsers.ts:256]: ohh I see, from the link?

hmmm... is it safe to leave it this way?

I still recommend patching up some random characters


you could create a new secret and put it in .env or slice from existing secrets

Comment (KristofaJosh) [reply apps/server/src/controllers/userControllers/authUsers.ts:256]: or

image


This way, we generate secret and long string id
when it's time to validate
We use the long string to pull out the data
then from that data, we get the secret
we then append the secret to the JWT existing access token secret
use that to validate the JWT, if the signature is valid, we proceed

File: apps/server/src/controllers/userControllers/authUsers.ts [hunk 2]
```diff
@@ -270,8 +276,9 @@ export const forgotPassword = async (req: Request, res: Response) => {
           message: `Password reset link will be sent to your email if you have an account with us.`,
         })
       }
```
Comment (KristofaJosh) [inline apps/server/src/controllers/userControllers/authUsers.ts:278]: I don't think we need this

**Review by KristofaJosh — APPROVED** [review]
LGTM!

File: apps/server/src/controllers/userControllers/authUsers.ts (hunk 2 above)
Comment (KristofaJosh) [reply apps/server/src/controllers/userControllers/authUsers.ts:278]: I think the last return will always be hit

**Pull request merged**